tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1756,7 +1817,9 @@ def grayscale_to_rgb(images, name=None):
   Returns:
     The converted grayscale image(s).
   """
-  with ops.name_scope(name, 'grayscale_to_rgb', [images]) as name:
+  with ops.name_scope(name, 'grayscale_to_rgb', [images]) as name:    
+    _AssertGrayscaleImage(images)
```

Comment:
Perhaps it's convenient for user to convert 2D to 3D automatically, and give a warning.

Reply:
I thought about this as well but finally decided against this. I think it's a better approach to stick to one fixed definition of grayscale images across the entire framework, i.e. a tensor with at least 3 dimensions and the last dimension is size 1.

Reply:
_AssertGrayscaleImage(images) doesn't do anything if there's no control dependency on its result

Reply:
@alextp You are right, the assert function was only working for cases where the shape could be inferred statically. This should be solved now.